fulp_modules/lisa/joel_gun.dm

```diff
@@ -0,0 +1,120 @@
+/obj/item/gun/ballistic/revolver/joel
+	icon = 'icons/obj/guns/ballistic.dmi'
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi' /// We probably don't want to change these...
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	load_sound = 'fulp_modules/lisa/Sounds/gunload.ogg'
+	eject_sound = 'fulp_modules/lisa/Sounds/empty.ogg'
+	icon_state = "revolver"
+	name = "\improper Bolt Action pistol"
+	desc = "The most powerful handgun in Olathe. It's best not to waste the only bullet."
+	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/c22
+	var/used_ability = FALSE
+
+/// To prevent ability spam.
+/obj/item/gun/ballistic/revolver/joel/proc/clear_cooldown()
+	used_ability = FALSE
+
+/// Load It In
+/obj/item/gun/ballistic/revolver/joel/attackby(obj/item/A, mob/user, params)
+	..()
+	if(!istype(A, /obj/item/ammo_casing))
+		return
+	user.visible_message("<span class='danger'>[user.name] puts the bullet in [src]!</span>",\
+	 "<span class='userdanger'>You put the bullet in [src]!</span>",\
+	 "<span class='hear'>You hear metal clanking...</span>")
+	var/mob/living/carbon/C = user
+	C.adjustBruteLoss(-10)
+	update_appearance()
+	A.update_appearance()
+	return
+
+/// Gun Reveal + Gun Toss
+/obj/item/gun/ballistic/revolver/joel/attack_self(mob/user)
+	if(!used_ability)
+		if(HAS_TRAIT(user, TRAIT_GUNFLIP)) /// If you have a holster, use gun reveal.
+			user.visible_message("<span class='danger'>[user.name] opens their holster and pulls out [src]!</span>",\
+			 "<span class='userdanger'>You reveal [src] from your holster!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/carbon/human/L in viewers(7, user))
+				if(prob(15) && !(L == user) && (L.stat))
+					to_chat(L, "<span class='warning'>Seeing [src] revealed in such a manner disgusts you!</span>")
+					if(!L.mind || !istype(L.mind.martial_art, /datum/martial_art/velvetfu))
+						L.vomit()
```

Comment:
please for the love of god make this use non-stunning vomit instead of stunning vomit. this is straight up just a 15% chance (every 5 seconds) to win a fight instantly as long as you're on someone's screen (and able to act, etc.).

Reply:
It already is the non stunning vomit, I've tested it in game.

Reply:
what did you test it on?

/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, vomit_type = VOMIT_TOXIC, harm = TRUE, force = FALSE, purge_ratio = 0.1)

stun is TRUE under the default parameters